Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+                }
+            }
+            
+            // Remove documented parameters that don't apply to this language representation's function signature.
+            var languageApplicableParameters = parametersByName.filter { knownParameterNames.contains($0.key) }
+            guard !languageApplicableParameters.isEmpty else { 
+                // Add an empty section so that this language doesn't fallback to another language's content.
+                variants[trait] = ParametersSection(parameters: [])
+                continue
+            }
+            
+            // Add a missing error parameter documentation if needed.
+            if trait == .objectiveC, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) {
+                languageApplicableParameters["error"] = [Parameter(name: "error", contents: Self.objcErrorDescription)] // This parameter is synthesized and doesn't have a source range.
+            }
+            
+            // Ensure that the parameters are displayed in the order that they're declared.
+            var sortedParameters: [Parameter] = []
+            sortedParameters.reserveCapacity(languageApplicableParameters.count)
+            for parameter in signature.parameters {
+                if let foundParameter = languageApplicableParameters[parameter.name]?.first {
+                    sortedParameters.append(foundParameter)
+                }
+            }
+            
+            variants[trait] = ParametersSection(parameters: sortedParameters)
+        }
+        
+        // Diagnose documented parameters that's not found in any language representation's function signature.
+        for parameter in parameters where !allKnownParameterNames.contains(parameter.name) {
+            if let matchingParameterNames = argumentLabelsAndPossibleParameterNames[parameter.name] {
+                diagnosticEngine.emit(externalParameterNameProblem(parameter, knownParameterNamesForArgumentLabel: matchingParameterNames.sorted()))
+            } else {
+                diagnosticEngine.emit(extraParameterProblem(parameter, knownParameterNames: allKnownParameterNames, symbolKind: symbolKind))
+            }
+        }
+        
+        // Diagnose parameters that are documented more than once.
+        for (name, parameters) in parametersByName where allKnownParameterNames.contains(name) && parameters.count > 1 {
+            let first = parameters.first! // Each group is guaranteed to be non-empty.
+            for parameter in parameters.dropFirst() {
+                diagnosticEngine.emit(duplicateParameterProblem(parameter, previous: first, symbolKind: symbolKind))
+            }
+        }
+        
+        // Diagnose missing parameters, based on the language representation with the most parameters.
+        if let parameterOrder = signatures.values.map(\.parameters).max(by: { $0.count < $1.count })?.map(\.name) {
```

Comment:
This line of code is especially confusing. Maybe add more explanation to the comment?

Also, the name `parameterOrder` doesn't help much to understand what this value is. If I'm reading this correctly, `parameterOrder` is an array of parameter names from the language/trait that has the most parameters.

A variable name such as `parameterNamesOrder` or `parameterNamesFromLongestSignature` or something similar would make it clear this is an array of names.

Reply:
Interesting - I prefer to try to name variables in a way that describes what they are/contain, and not what they are used for... The following lines of code will show the code reader what it is used for.

Reply:
I documented the assumptions about why the language representation with the most parameters can determine the order in other language representations as well in https://github.com/apple/swift-docc/pull/776/commits/ef264671245be85daf65bb1ab7a3eb96a6d8a95b